frontend/src/Pages/Login.js

```diff
@@ -1,86 +1,179 @@
-import React, { useState } from 'react'
+import React, { useEffect, useState } from 'react'
 import { Link } from 'react-router-dom'
 import PropTypes from "prop-types"
 
-function AuthButton({ imgUrl }){
-	return (
-		<button
-			className="block h-8 w-8 mx-4 rounded-full overflow-hidden border-2 border-gray-300 hover:border-white"
-			type="button"
-			style={{ transition: "all .15s ease" }}
-		>
-			<img
-			alt="..."
-			className="h-full w-full mr-1"
-			src={imgUrl}
-			/>
-		</button>
-	)
-}
-
-AuthButton.propTypes = {
-	imgUrl: PropTypes.string
-}
-
-function UserField({ type }) {
-	switch (type) {
-		case 'Username': {
-			return (
-				<div className="w-full mb-3">
-					<input
-					type="username"
-					className="border-0 px-3 py-3 placeholder-gray-400 text-gray-700 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full"
-					placeholder="Username"
-					style={{ transition: "all .15s ease" }}
-					/>
-				</div>
-			)
-		}
+import UserField from '../Components/UserField'
+import SSOLogin from '../Components/SSOLogin'
+import {parsePhoneNumber} from "react-phone-number-input"
+import validateAndReturnUrl from '../HelperFunctions/validateAndReturnUrl'
+import axios from 'axios'
 
-		case 'Email address': {
-			return (
-				<div className="w-full mb-3">
-					<input
-					type="email"
-					className="border-0 px-3 py-3 placeholder-gray-400 text-gray-700 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full"
-					placeholder="Email"
-					style={{ transition: "all .15s ease" }}
-					/>
-				</div>
-			)
+
+export default function Login() {
+	const default_homeserver = 'matrix.org';
+
+	// Used to set the Identifier type for password based login
+	const [fieldType, setFieldType] = useState('Username');
+	const [userField, setUserField] = useState('');
+
+	// State variable to store the complete phone number including the country calling code with no '+'.
+	//Ex: 919676765xxxxx (India (91) )
+	const [phoneNumber, setPhoneNumber] = useState('');
+
+	const [password, setPassword] = useState('');
+
+	// Updated every time the homeserver input text is changed.
+	const [inputHomeServer, setInputHomeServer] = useState('matrix.org');
+
+	// Updated after save button is clicked
+	const [homeServer, setHomeServer] = useState(default_homeserver);
+
+	// List of available SSO providers
```

Comment:
These comments might be a little redundant, since the variable names cover the usage :). I'd only include a comment if it was non-obvious what the variable was for.